transports/quic/src/config.rs

```diff
@@ -128,7 +132,32 @@ impl From<Config> for QuinnConfig {
         let mut client_config = quinn::ClientConfig::new(client_tls_config);
         client_config.transport_config(transport);
 
-        let mut endpoint_config = quinn::EndpointConfig::default();
+        let mut endpoint_config = match keypair.key_type() {
+            libp2p_identity::KeyType::RSA => {
+                log::warn!(
+                    "identity keytype is RSA, QUIC stateless resets will not be not supported"
+                );
+                quinn::EndpointConfig::default()
+            }
+            libp2p_identity::KeyType::Ed25519 => {
+                let keypair = keypair.try_into_ed25519().expect("key should be Ed25519");
+                let secret = keypair.secret();
+                endpoint_config(secret.as_ref())
```

Comment:
If we use `unwrap()` instead and inline all the variables, can this be a one-liner?